services/QuillLMS/app/controllers/teachers/units_controller.rb

```diff
@@ -215,13 +215,9 @@ def score_info
 
     return [] if teach_own_or_coteach_classrooms_array.empty?
 
-    scores, completed, archived_activities = ''
+    scores, completed = ''
 
-    if report
-      completed = lessons ? "HAVING ca.completed" : "HAVING SUM(CASE WHEN act_sesh.visible = true AND act_sesh.state = 'finished' THEN 1 ELSE 0 END) > 0"
-    else
-      archived_activities = "AND 'archived' != ANY(activities.flags)"
-    end
+    completed = lessons ? "HAVING ca.completed" : "HAVING SUM(CASE WHEN act_sesh.visible = true AND act_sesh.state = 'finished' THEN 1 ELSE 0 END) > 0" if report
```

Comment:
While I normally prefer guard clauses,  I think a conditional block might be more readable given the ternary as well as expression length:

```
if report
  completed = lessons ? "HAVING ca.completed" : "HAVING SUM(CASE WHEN act_sesh.visible = true AND act_sesh.state = 'finished' THEN 1 ELSE 0 END) > 0"
end
```